pkg/chrootarchive/archive_unix.go

```diff
@@ -79,6 +80,11 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 	w.Close()
 
 	if err := cmd.Wait(); err != nil {
+		// when `xz -d -c -q | docker-untar ...` failed on docker-untar side,
+		// we need to exhaust `xz`'s output, otherwise the `xz` side will be
+		// pending on write pipe forever
+		io.Copy(ioutil.Discard, decompressedArchive)
```

Comment:
better to use the same file `stage3-x32-20151004.tar.xz` as I am testing, is large enough, and will fail extracting xattr if running on aufs
http://mirrors.kernel.org/gentoo/releases/amd64/autobuilds/current-stage3-x32/